### PR TITLE
Improve ClientRequestSizeMetrics test

### PR DESCRIPTION
**Pull request by blerer**
The test was only checking that the metrics where updated but was not
testing if the updates were correct.

This patch also fix the ClearableHistogram.clear method (through an ugly
hack) that was not clearing the histogram count.

### CODE REVIEW

File: test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        clearMetrics();
+
+        executeNet("SELECT * from system.peers");
+        assertThat(ClientRequestSizeMetrics.totalBytesRead.getCount()).isGreaterThan(0);
+        assertThat(ClientRequestSizeMetrics.totalBytesWritten.getCount()).isGreaterThan(0);
+
+        // The request fit in one single frame so we know the new number of received frames
+        assertEquals(1, ClientRequestSizeMetrics.bytesReceivedPerFrame.getCount());
+
+        long requestLength = ClientRequestSizeMetrics.totalBytesRead.getCount();
```
Comment (bereng) [inline test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java:57]: Can't you declare these variables at line 56 and use them for those 2 initial asserts?

File: test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        assertMax(snapshot, requestLength);
+
+        // The response fit in one single frame so we know the new number of received frames
+        assertEquals(1, ClientRequestSizeMetrics.bytesTransmittedPerFrame.getCount());
+
+        snapshot = ClientRequestSizeMetrics.bytesTransmittedPerFrame.getSnapshot();
+        assertMin(snapshot, responseLength);
+        assertMax(snapshot, responseLength);
+
+        // Let's fire the same request again and test that the changes are the same that previously
```
Comment (bereng) [inline test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java:65]: Can't you avoid duplicating code by folding into a loop such as:
```
for (int i=1; i <=2 ; i++)
{
  long expectedTotalBytesRead = i * requestLength;
  ....
}
```

File: test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java (hunk 1 above)
Comment (blerer) [reply test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java:57]: Done :-)

File: test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java (hunk 2 above)
Comment (blerer) [reply test/unit/org/apache/cassandra/metrics/ClientRequestSizeMetricsTest.java:65]: There is some differences between the 2 set of checks so it is not really feasable to use a loop but I can extract a method to reduce the code duplication.

**Comment by bereng** [comment]
+1

**Pull request closed without merging**